enclave/src/openfinex/fixed_point_number_converter.rs

```diff
@@ -0,0 +1,256 @@
+// This file is part of Polkadex.
+
+// Copyright (C) 2020-2021 Polkadex oü.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+pub extern crate alloc;
+
+use crate::constants::UNIT;
+use crate::openfinex::openfinex_api::{OpenFinexApiError, OpenFinexApiResult};
+use crate::openfinex::openfinex_types::ResponseInteger;
+use alloc::{string::String, string::ToString, vec::Vec};
+use core::iter::Peekable;
+
+/// Provides conversion from and to a string containing a fixed point decimal with 18-digit precision
+/// to an u128 integer, scaled by UNIT (=1e18).
+pub struct FixedPointNumberConverter {}
```

Comment:
This is the new converter, i.e. parser and serializer, for fixed point number strings.
Replaces the previous implementation in `number_value_conversion.rs`